pvlib/modelchain.py

```diff
@@ -931,23 +996,125 @@ def prepare_inputs(self, weather, times=None):
+            include ``'poa_global'``, ``'poa_direct'`` and ``'poa_diffuse'``.
+            Columns with weather-related data are ssigned to the
+            ``weather`` attribute.  If columns for ``'temp_air'`` and
+            ``'wind_speed'`` are not provided, air temperature of 20 C and wind
+            speed of 0 m/s are assumed.
+
+        Notes
+        -----
+        Assigns attributes: ``weather``, ``total_irrad``, ``solar_position``,
+        ``airmass``, ``aoi``, ``temperature``.
+
+        See also
+        --------
+        pvlib.modelchain.ModelChain.prepare_inputs
+        """
+
+        self._assign_weather(data)
+
+        self._verify_df(data, req=['poa_global', 'poa_direct', 'poa_diffuse'])
+        self._assign_total_irrad(data)
+
+        self._prep_inputs_solar_pos()
+        self._prep_inputs_airmass()
+
+        if isinstance(self.system, SingleAxisTracker):
+            self._prep_inputs_tracking()
+        else:
+            self._prep_inputs_fixed()
+
+        return self
+
+    def prepare_temperature(self, data=None):
+        """
+        Sets cell_temperature using inputs in data and the specified
+        temperature model.
+
+        If 'data' contains 'cell_temperature', these values are assigned to
+        attribute ``cell_temperature``. If 'data' contains 'module_temperature`
+        and `temperature_model' is 'sapm', cell temperature is calculated using
+        :py:func:`pvlib.temperature.sapm_celL_from_module`. Otherwise, cell
+        temperature is calculated by 'temperature_model'.
+
+        Parameters
+        ----------
+        data : DataFrame, default None
+            May contain columns ``'cell_temperature'`` or
+            ``'module_temperaure'``.
+
+        Returns
+        -------
+        self
+
+        Assigns attribute ``cell_temperature``.
+
+        """
+        try:
+            self.cell_temperature = data['cell_temperature']
+            if self.temperature_model is not None:
+                warnings.warn('using ''cell_temperature'' provided in input',
+                              ' rather than specified temperature model: {0}'
+                              .format(self.temperature_model.__name__))
+            return self
+        except:
+            pass
+
+        # cell_temperature is not in input. Calculate cell_temperature using
+        # a temperature_model.
+        # If module_temperature is in input data we can use the SAPM cell
+        # temperature model.
+        if ('module_temperature' in data) and \
+            (self.temperature_model.__name__ == 'sapm'):
+            # use SAPM cell temperature model only
+            self.cell_temperature = pvlib.temperature.sapm_cell_from_module(
+                module_temperature=data['module_temperature'],
+                poa_global=self.total_irrad['poa_global'],
+                deltaT=self.temperature_model_parameters['deltaT'])
+            return self
```

Comment:
If I understand correctly, we can remove this `return self` and add `else: self.temperature_model()`

Reply:
If `self.temperature_model` needs some more data then that method can complain about it.

Reply:
I agree but we need to resolve the [issue](https://github.com/pvlib/pvlib-python/pull/943#discussion_r482330580) with the effective irradiance entrypoint first